### ISSUE TITLE
Do I need P_fac, A_fac, L_fac for image-level output?

### ISSUE THREAD

**Issue by whqwill** [DirtyHarryLYL/HAKE-Action#65]
I noticed the Generate_detection.py that there are P_fac, A_fac, L_fac for factorize score_P, score_A and score_L. Also, there is sigmoid function for calculating score_P, score_A and score_L. But in image-level output, there is no sigmoid function and factorization. Do I need them as well?

**Comment by whqwill** [comment]
And why do you use torch.sigmoid again when calculating sP, sA and sL?

**Comment by Shunli-Wang** [comment]
I have the same question. And I can't figure it out whether we need to joint train with TIN and A2V. If yes, I can't find the joint training code. If no, how does this fuse command come from?
`sHO = (((sH + sO) * ssp + sP + sA + sL) * score_I[im_index[keys[obj_index]], x:y]) * hod`

**Comment by Shunli-Wang** [comment]
How does `generation_args.pkl` comes from?  Why we need to divide scores by `P_fac, A_fac, L_fac` separately? 
`score_P[obj_index][:, hoi_index - x]  /= P_fac[hoi_index]`
`score_A[obj_index][:, hoi_index - x]  /= A_fac[hoi_index]`
`score_L[obj_index][:, hoi_index - x]  /= L_fac[hoi_index]`

**Comment by Foruck** [comment]
> I noticed the Generate_detection.py that there are P_fac, A_fac, L_fac for factorize score_P, score_A and score_L. Also, there is sigmoid function for calculating score_P, score_A and score_L. But in image-level output, there is no sigmoid function and factorization. Do I need them as well?

No.

**Comment by Foruck** [comment]
> And why do you use torch.sigmoid again when calculating sP, sA and sL?

Note that the saved sP, sA and sL are unbounded logits. We use sigmoid to project them to the probability space.

**Comment by Foruck** [comment]
> How does `generation_args.pkl` comes from? Why we need to divide scores by `P_fac, A_fac, L_fac` separately?
> `score_P[obj_index][:, hoi_index - x] /= P_fac[hoi_index]`
> `score_A[obj_index][:, hoi_index - x] /= A_fac[hoi_index]`
> `score_L[obj_index][:, hoi_index - x] /= L_fac[hoi_index]`

For the two questions, please take reference from #30 .

**Comment by Foruck** [comment]
Duplicate issue with #30 .
